test-data/unit/check-protocols.test

```diff
@@ -2536,3 +2536,38 @@ class EmptyProto(Protocol): ...
 def hh(h: EmptyProto) -> None: pass
 hh(None)
 [builtins fixtures/tuple.pyi]
+
+
+[case testPartialTypeProtocol]
+from typing import Protocol
+
+class Flapper(Protocol):
+    def flap(self) -> int: ...
+
+class Blooper:
+    flap = None
+
+    def bloop(self, x: Flapper) -> None:
+        reveal_type([self, x])  # N: Revealed type is 'builtins.list[builtins.object*]'
+
+class Gleemer:
+    flap = []  # E: Need type annotation for 'flap' (hint: "flap: List[<type>] = ...")
+
+    def gleem(self, x: Flapper) -> None:
+        reveal_type([self, x])  # N: Revealed type is 'builtins.list[builtins.object*]'
+[builtins fixtures/tuple.pyi]
+
+
+[case testPartialTypeProtocolHashable]
+# flags: --strict-optional
```

Comment:
Do you need this? I thought it was the default. Or are the tests run multiple times with different default flags?

Reply:
Oh, I meant to use `--no-strict-optional` as in https://github.com/python/mypy/issues/9437#issuecomment-699788746 — but a test a couple lines up used `--strict-optional` and I copied it by accident. Thanks!

Hmm. It actually looks like `--no-strict-optional` is the default in tests. I just discovered this file lurking in the shadows: https://github.com/python/mypy/blob/8bf770d63cc4ac656dde00904b61d7f1b6fba68e/test-data/unit/hacks.txt#L8

Also `object` in the test stubs doesn't have `__hash__`... so this test didn't actually test anything different from Blooper (and crashes regardless of `--strict-optional`). Fixed now (ie, the new test and old code only crashes with `--no-strict-optional`).